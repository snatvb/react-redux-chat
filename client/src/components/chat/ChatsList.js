/**
 * Created by snatvb on 17.12.16.
 */

import React, { Component } from 'react';
import {connect} from 'react-redux';
//import * as usersActions from '../../actions/users';
//import * as profileActions from '../actions/profile';
import * as chatsActions from '../../actions/chats';

import ChatsListItem from './ChatsListItem';

//@connect(store => {
//    return {
//        //users: store.users.users,
//        //userFetched: store.users.fetched,
//        chats: store.chats.chats,
//        chatsFetched: store.chats.fetched,
//        profile: store.profile.profile,
//        profileFetched: store.profile.fetched
//    }
//})
class ChatsList extends Component {
    componentWillReceiveProps(nextProps) {
        //nextProps.users.forEach(user => {
        //    this.props.dispatch(chatsActions.addChat(user.id, user));
        //});
        if (!nextProps.chatsFetched) {
            nextProps.dispatch(chatsActions.setChats(nextProps.users));
        }
    }

    renderChatsList() {
        const {chats, profile, selected} = this.props;
        let usersItems = [];
        chats.forEach(chat => {
            //console.log(chat.user.id === profile.userId);
            if (chat.user.id === profile.userId) {
                return null;
            }
            const isSelected = parseInt(selected, 10) === chat.user.id;
            usersItems.push(<ChatsListItem key={chat.id} selected={isSelected} chat={chat} />)
        });
        return usersItems;
    }


    render() {
        let chats = this.renderChatsList();
        return (
            <div className="chat-list">
                {chats}
            </div>
        );
    }
}

ChatsList.propTypes = {
    users: React.PropTypes.array.isRequired
};

export default connect(store => {
    return {
        //users: store.users.users,
        //userFetched: store.users.fetched,
        chats: store.chats.chats,
        chatsFetched: store.chats.fetched,
        profile: store.profile.profile,
        profileFetched: store.profile.fetched
    }
})(ChatsList);
export const undecorated = ChatsList;