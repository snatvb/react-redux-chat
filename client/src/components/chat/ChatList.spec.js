/**
 * Created by snatvb on 19.12.16.
 */

import React from 'react';
import {undecorated as ChatList} from './ChatsList';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';


describe('Chats list component', () => {
    it('render list chats', () => {
        const props = {
            users: [
                {
                    id: 23432,
                    name: "tester"
                },
                {
                    id: 23435,
                    name: "tester 2",
                    online: true
                }
            ],
            chats: [
                {
                    id: 23432,
                    user: {
                        id: 23432,
                        name: "tester"
                    },
                    messages: []
                },
                {
                    id: 23435,
                    user: {
                        id: 23435,
                        name: "tester 2",
                        online: true
                    },
                    messages: []
                }
            ],
            chatsFetched: true,
            profile: {
                userName: 'TESTER 4',
                id: 2352342
            },
            profileFetched: true
        };
        const component = renderer.create(
            <ChatList {...props} />
        );
        expect(component).toMatchSnapshot();
    });
});