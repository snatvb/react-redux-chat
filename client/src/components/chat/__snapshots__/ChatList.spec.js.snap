exports[`Chats list component render list chats 1`] = `
<div
  className="chat-list">
  <div
    className="chat-list_item"
    onClick={[Function]}>
    tester
  </div>
  <div
    className="chat-list_item"
    onClick={[Function]}>
    tester 2
    <div
      className="chat-list_item_online">
      Онлайн
    </div>
  </div>
</div>
`;
